pilot/pkg/bootstrap/certcontroller.go

```diff
@@ -129,10 +130,22 @@ func (s *Server) initDNSCerts(hostname, customHost, namespace string) error {
 
 	var certChain, keyPEM, caBundle []byte
 	var err error
-	if features.PilotCertProvider == constants.CertProviderKubernetes {
+	if features.PilotCertProvider == constants.CertProviderKubernetes && signerName != "" && s.RA != nil {
```

Comment:
Can you add comments? Its not 100% clear why we have this complex set of logic to determine this now - so in a year it will probably be confusing

Reply:
```
  pilot:
    k8s:
      env:
      - name: EXTERNAL_CA
        value: ISTIOD_RA_KUBERNETES_API
      - name: K8S_SIGNER
        value: kubernetes.io/legacy-unknown
      - name: PILOT_CERT_PROVIDER
        value: kubernetes
```
Take above config as an example,  we do not need to enable ca function in istiod at all since we will delegate all certs generation to the external ca and istiod is just working as an RA.

Reply:
This check isn't entirely accurate. It is still acceptable to use the Kubernetes cert provider even if s.RA is nil. Also, this creates a lot of backward compatibility issues by FORCING a signer name to be present, since K8S_SIGNER is still an experimental config whereas using K8s to sign certs is already GA. I would suggest removing the newly added checks

Reply:
In general, IMO, s.RA (and its corresponding config which is EXTERNAL_CA) should only be used to determine the workload cert signer, not the control-plane cert signer. @irisdingbj - WDYT?

Reply:
Agree. We can extend this in the future to allow user define signer for istiod certs as well via introducing a new Env Var:
`PILOT_CERT_SIGNER`